eawx-build-test/Configuration/v1/BuildConfigParserTest.cs

```diff
@@ -0,0 +1,546 @@
+            }
+        }
+
+        [TestMethod]
+        public void GivenXmlWithSingleProject__WhenCallingParse__ShouldReturnArrayWithMatchingProject()
+        {
+            const string xml = @"<?xml version=""1.0"" encoding=""UTF-8""?>
+                                <eaw-ci:BuildConfiguration
+                                  ConfigVersion=""1.0.0"" 
+                                  xmlns:eaw-ci=""eaw-ci""
+                                  xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"">
+                                  <Projects>
+                                    <Project Id=""TestProject"">
+                                      <Jobs>
+                                        <Job Id=""idvalue1"">
+                                          <Tasks>
+                                          </Tasks>
+                                        </Job>
+                                      </Jobs>
+                                    </Project>
+                                  </Projects>
+                                </eaw-ci:BuildConfiguration>";
+
+            _mockFileData.TextContents = xml;
+
+            const string projectName = "TestProject";
+
+            var factoryStub = new BuildComponentFactoryStub { Project = new ProjectStub() };
+            var sut = new BuildConfigParser(_fileSystem, factoryStub);
+
+            var projects = sut.Parse(Path);
+
+            var actual = projects.ToList()[0] as ProjectStub;
+            var actualName = actual?.Name;
+            AssertProjectNameEquals(projectName, actualName);
+        }
+
+        [TestMethod]
+        public void GivenXmlWithSingleProjectAndJob__WhenCallingParse__ProjectShouldHaveMatchingJob()
+        {
+            const string xml = @"<?xml version=""1.0"" encoding=""UTF-8""?>
+                                <eaw-ci:BuildConfiguration
+                                  ConfigVersion=""1.0.0"" 
+                                  xmlns:eaw-ci=""eaw-ci""
+                                  xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"">
+                                  <Projects>
+                                    <Project Id=""idvalue0"">
+                                      <Jobs>
+                                        <Job Id=""idvalue1"" Name=""TestJob"">
+                                          <Tasks>
+                                          </Tasks>
+                                        </Job>
+                                      </Jobs>
+                                    </Project>
+                                  </Projects>
+                                </eaw-ci:BuildConfiguration>";
+
+            _mockFileData.TextContents = xml;
+
+            const string jobName = "TestJob";
+
+            var factoryStub = new BuildComponentFactoryStub { Project = new ProjectStub(), Job = new JobStub() };
+            var sut = new BuildConfigParser(_fileSystem, factoryStub);
```

Comment:
Not a necessity, but what we could look into is dependency injection, so instead of calling `new Type(stuff)` we could use something like `var svc = Services.get<IType>();`

Reply:
Do we have to pass the `ServiceProvider` to all the classes manually or is there a magic annotation that resolves everything for us?
I'm not entirely sure we need it since `Tasks` are built by the `TaskBuilders` created in `BuildComponentFactory`, so we have all the necessary dependencies in one place anyway.

Reply:
Depends on the library, but usually you allow the required elements to be passed to the constructor of an object, but don'T have to pass the service provider itself:
https://autofaccn.readthedocs.io/en/latest/getting-started/index.html